x-pack/plugins/apm/public/components/shared/suggestions_select/suggestions_select_with_term.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiComboBox, EuiComboBoxOptionOption } from '@elastic/eui';
+import { debounce } from 'lodash';
+import React, { useCallback, useState } from 'react';
+import { FETCH_STATUS, useFetcher } from '../../../hooks/use_fetcher';
+
+interface SuggestionsSelecttWithTermProps {
+  allOption?: EuiComboBoxOptionOption<string>;
+  customOptionText?: string;
+  defaultValue?: string;
+  field: string;
+  termField: string;
+  termValue: string;
+  start?: string;
+  end?: string;
+  onChange: (value?: string) => void;
+  placeholder: string;
+  prepend?: string;
+}
+
+export function SuggestionsSelectWithTerm({
+  allOption,
+  customOptionText,
+  defaultValue,
+  field,
+  termField,
+  termValue,
+  start,
+  end,
+  onChange,
+  placeholder,
+  prepend,
+}: SuggestionsSelecttWithTermProps) {
+  const allowAll = !!allOption;
+  let defaultOption: EuiComboBoxOptionOption<string> | undefined;
+
+  if (allowAll && !defaultValue) {
+    defaultOption = allOption;
+  }
+  if (defaultValue) {
+    defaultOption = { label: defaultValue, value: defaultValue };
+  }
+  const [selectedOptions, setSelectedOptions] = useState(
+    defaultOption ? [defaultOption] : []
+  );
+
+  const [searchValue, setSearchValue] = useState('');
+
+  const { data, status } = useFetcher(
+    (callApmApi) => {
+      return callApmApi('GET /internal/apm/suggestions_with_terms', {
+        params: {
+          query: {
+            field,
+            string: searchValue,
+            start,
+            end,
+            termField,
+            termValue,
+          },
+        },
+      });
+    },
+    [field, searchValue],
+    { preservePreviousData: false }
+  );
+
+  const handleChange = useCallback(
+    (changedOptions: Array<EuiComboBoxOptionOption<string>>) => {
+      setSelectedOptions(changedOptions);
+      if (changedOptions.length === 1) {
+        onChange(
+          changedOptions[0].value
+            ? changedOptions[0].value.trim()
+            : changedOptions[0].value
+        );
+      }
+    },
+    [onChange]
+  );
+
+  const handleCreateOption = useCallback(
+    (value: string) => {
+      handleChange([{ label: value, value }]);
+    },
+    [handleChange]
+  );
+
+  const terms = data?.terms ?? [];
+
+  const options: Array<EuiComboBoxOptionOption<string>> = [
+    ...(allOption &&
+    (searchValue === '' ||
+      searchValue.toLowerCase() === allOption.label.toLowerCase())
+      ? [allOption]
+      : []),
+    ...terms.map((name) => {
+      return { label: name, value: name };
+    }),
+  ];
+
+  return (
+    <EuiComboBox
+      async={true}
+      compressed={true}
```

Comment:
@kpatticha Any way we can pass this as `false` in the case of the environment selector on the service views? We want the control to be the default size, not compact.

I imagine this is the same component in use in the anomaly detection job creation?
<img width="538" alt="CleanShot 2022-03-02 at 20 17 40@2x" src="https://user-images.githubusercontent.com/4104278/156433017-8648bd41-cf7b-4361-9e35-9df5b103b3fb.png">

Reply:
For `EuiComboBox` the `compressed` default value is `false` so I'm removing it. We can customise later or if needed. 

Both are using the `EuiComboBox` component.